apps/system/js/app_chrome.js

```diff
@@ -340,6 +373,67 @@
     window.dispatchEvent(new CustomEvent('global-search-request'));
   };
 
+  AppChrome.prototype.onClickSiteIcon = function onClickSiteIcon() {
+    this.pinningPref && this.pinningPref.get(function(enabled) {
+      if (!enabled || !this.app.isBrowser()) {
```

Comment:
We need to figure out how to make this work for app windows too. But I'm not sure how yet...